users/models.py

```diff
@@ -29,8 +34,15 @@ def _create_user(self: "UserManager", email: str, nickname: str, password: str,
         email = self.normalize_email(email)
         user = self.model(email=email, nickname=nickname, **extra_fields)
         user.set_password(password)
-        user.save(using=self._db)
-        return user
+        try:
+            with transaction.atomic():
+                user.save(using=self._db)
+                return user
+        except IntegrityError:
+            user.nickname = nickname + generate_secure_random_digits()
+            with transaction.atomic():
+                user.save(using=self._db)
+                return user
```

Comment:
Too many returns. Let's refactor a bit
```suggestion
        with transaction.atomic():
            try:
                user.save()
            except IntegrityError:
                user.nickname = nickname + generate_secure_random_digits()
                user.save()
        return user
```
You may need to adjust indentation.

Reply:
Actually, is transaction.atomic needed here? Since we are saving single model, it is not necessary. Atomic transactions are useful if we would have multiple operations that needs to be saved together. So IMO - drop this atomic transaction.